### PR TITLE
Added an example for running KinD on GitLab

### PR DESCRIPTION
**Pull request by shreyu86**

### CODE REVIEW

**Comment by aojea** [comment]
Do the users need to do additional changes on the host, like mounting some folders?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+test:
+  stage: test
+  script:
+    - apk add -U curl
+    - curl -Lo ./kind-linux-amd64 https://github.com/kubernetes-sigs/kind/releases/download/v0.4.0/kind-linux-amd64
+    - chmod +x ./kind-linux-amd64
+    - mv ./kind-linux-amd64 /usr/local/bin/kind
+    - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+    - chmod +x ./kubectl
+    - ./kubectl get nodes -o wide
```
Comment (tao12345666333) [inline .gitlab-ci.yml]: In fact, this doesn't work.You didn't create a cluster

**Comment by tao12345666333** [comment]
I think in the example, it would be better to let `./kind-smoketest.sh` run successfully.

**Comment by shreyu86** [comment]
@aojea @tao12345666333  thanks for the feedback, last evening I might have checked in partial version of my .gitlab-ci.yml file. I will make changes and put those up for review tonight.

**Comment by shreyu86** [comment]
@aojea @tao12345666333 the tests now pass on Gitlab.com on my fork. Thanks a lot to #11  and @phisco's pull request which helped me solve https://github.com/kubernetes-sigs/kind/issues/630

File: gitlab/kind-config.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: kind.sigs.k8s.io/v1alpha3
+kind: Cluster
+networking:
+  apiServerAddress: "0.0.0.0"
```
Comment (BenTheElder) [inline gitlab/kind-config.yaml]: I really hope this doesn't mean it binds to any public interfaces :sweat_smile:

**Comment by BenTheElder** [comment]
thanks!

**Pull request merged**